go/base/tasks.py

```diff
@@ -0,0 +1,49 @@
+import json
+from celery.task import task
+
+from go.base.utils import vumi_api
+from go.config import get_conversation_definition
+from go.vumitools.metrics import get_conversation_metric_prefix
+
+
+@task(ignore_result=True)
+def send_recent_conversation_metrics():
+    api = vumi_api()
+    try:
+        conversation_details = get_and_reset_recent_conversations(api)
+
+        for conv_details in conversation_details:
+            details = json.loads(conv_details)
+            user_api = api.get_user_api(details["account_key"])
+            conv = user_api.get_conversation(details["conv_key"])
+            publish_conversation_metrics(api, user_api, conv)
+    finally:
+        api.close()
+
+
+def get_and_reset_recent_conversations(vumi_api):
+    redis = vumi_api.redis.sub_manager("conversation.metrics.middleware")
+
+    # makes use of redis atomic functions to ensure nothing is added to the set
+    # before it is deleted
+    try:
+        redis.rename("recent_coversations", "old_recent_conversations")
+    except:
+        return []
```

Comment:
I think this might also work:
`from redis.exceptions import ResponseError as redis_response_error
from vumi.persist.fake_redis import ResponseError as fake_response_error`
and then
`except (redis_response_error, fake_response_error):`

I also prefer the first method but there is this line in the ticket "The idea will be to only fire the metrics once a day (or hour, or every 100 messages)." If they want to change the regularity at some point do you think the atomic-ness will become a requirement?